docs/source/architecture/namespace_permission.rst

```diff
@@ -0,0 +1,46 @@
+***********************
+Namespace Permissioning
+***********************
+
+A transaction family is defined by a set of parameters including a
+name, some version numbers, an *apply* function and namespaces. The
+apply function is what defines the behavior of a transaction family
+according to the `get` and `set` operations on addresses of the Global
+State. The goal of having a family declaring namespaces is to shard
+the set of addresses in order to avoid transaction families tampering
+one another's addreses. More precisely, a namespace consists of an
+address prefix, usually 3 bytes.  When desired, namespace
+permissioning is enforced by the validators by ensuring that all `set`
+operations performed by a given transaction family write only to
+addresses with the family’s specified namespace prefix(es). By default
+and for better flexibility, this namespace permissioning is not
+enforced by the validators, but there are some scenarios where
+enforcement would increase the security of the execution of
+transaction families.  This would be the case in a scenario where the
+deployment of transaction families is open.
+
+In order to make validators enforce namespace permissions,
+appropriate settings must be published on-chain using the
+:doc:`Settings Transaction Family
+<../transaction_family_specifications/settings_transaction_family>`.
+For every block, a validator loads the json data stored under the
+specific key ``sawtooth.validator.transaction_families``.  At this
+key the data is a list of json entries, one per transaction family,
+possibly including its namespaces.
+
+Here is an example in which the namespace permissions are set for the
+transaction families ``block_info``, ``sawtooth_settings`` and
+``intkey``:
+
+.. code-block:: python
+
+  sawtooth.validator.transaction_families='[
+    {\"family\": \"block_info\", \"version\": \"1.0\", \"namespaces\": [\"00b10c\"]},
+    {\"family\":\"sawtooth_settings\", \"version\":\"1.0\"},
+    {\"family\": \"intkey\", \"version\": \"1.0\", \"namespaces\": [\"1cf126\"]} ]'
+
+In the case where a family has no ``"namespace"``, the validator does
+not restrict the write permission of the family. In the example above,
+the family `sawtooth_settings` does not indicate any namespace and
+therefore the validator will let it write at any address.
```

Comment:
... to any address defined by the intersection of by the namespace(s) present in the transaction processor's registration declaration and the output field of the transaction it is processing.

Reply:
I am not sure about that one. Some somewhat recent experiment make me believe that the transaction processor's registration declaration is not very helpful for the validator. The transaction processor could give some namespace at registration and later validate transactions not in these namespaces. Isn't it?

Separately, should "intersection" be "union"?